app/uk/gov/hmrc/cgtpropertydisposals/connectors/returns/SubmitReturnsConnector.scala

```diff
@@ -0,0 +1,261 @@
+    extends SubmitReturnsConnector
+    with DesConnector {
+
+  override def submit(
+    returnRequest: SubmitReturnRequest
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, HttpResponse] = {
+
+    val baseUrl: String    = config.baseUrl("returns")
+    val cgtReferenceNumber = returnRequest.subscribedDetails.cgtReference.value
+
+    val returnUrl: String = s"$baseUrl/cgt-reference/$cgtReferenceNumber/return"
+
+    val desSubmitReturnRequest = DesSubmitReturnRequest(returnRequest)
+
+    EitherT[Future, Error, HttpResponse](
+      http
+        .post(returnUrl, Json.toJson(desSubmitReturnRequest), headers)(
+          implicitly[Writes[JsValue]],
+          hc.copy(authorization = None),
+          ec
+        )
+        .map(Right(_))
+        .recover { case e => Left(Error(e)) }
+    )
+  }
+
+}
+
+object SubmitReturnsConnectorImpl {
+
+  final case class DesSubmitReturnRequest(ppdReturnDetails: PPDReturnDetails)
+
+  object DesSubmitReturnRequest {
+
+    def apply(submitReturnRequest: SubmitReturnRequest): DesSubmitReturnRequest = {
+      val ppdReturnDetails = getPPDReturnDetails(submitReturnRequest)
+      DesSubmitReturnRequest(ppdReturnDetails)
+    }
+
+    implicit val desSubmitReturnRequestFormat: OFormat[DesSubmitReturnRequest] = Json.format[DesSubmitReturnRequest]
+  }
+
+  private def getPPDReturnDetails(submitReturnRequest: SubmitReturnRequest): PPDReturnDetails = {
+    val c                      = submitReturnRequest.completeReturn
+    val returnDetails          = getReturnDetails(submitReturnRequest)
+    val lossSummaryDetails     = getLossSummaryDetails(c)
+    val reliefDetails          = getReliefDetails(c)
+    val incomeAllowanceDetails = getIncomeAllowanceDetails(c)
+    val disposalDetails        = getDisposalDetails(c)
+
+    PPDReturnDetails(
+      returnType = CreateReturnType(
+        source         = getSource(submitReturnRequest),
+        submissionType = SubmissionType.New
+      ),
+      returnDetails            = returnDetails,
+      representedPersonDetails = None,
+      disposalDetails          = disposalDetails,
+      lossSummaryDetails       = lossSummaryDetails,
+      incomeAllowanceDetails   = incomeAllowanceDetails,
+      reliefDetails            = reliefDetails
+    )
+  }
+
+  private def getSource(submitReturnRequest: SubmitReturnRequest): String =
+    submitReturnRequest.agentReferenceNumber.fold("self digital")(_ => "agent digital")
+
+  private def getAddresssDetails(c: CompleteReturn) =
+    AddresssDetails(
+      addressLine1 = c.propertyAddress.line1,
+      addressLine2 = c.propertyAddress.line2,
+      addressLine3 = c.propertyAddress.town,
+      addressLine4 = c.propertyAddress.town,
+      countryCode  = c.propertyAddress.countryCode,
+      postalCode   = c.propertyAddress.postcode
+    )
+
+  private def getDisposalDetails(c: CompleteReturn) = {
```

Comment:
one thing you might consider is moving some of these `getX` methods into the companion objects of the case class you're constructing as `apply` methods to spread out the code a bit